Yoast/wordpress-seo
# Fix template variable issue in twitter class, add unit tests

## Summary

This PR can be summarized in the following changelog entry:

* Properly replace snippet variables in Twitter card title and description.

## Test instructions

This PR can be tested by following these steps:

* Create a post that uses snippet variables in its twitter card title and description, see them work.
* Create a category / tag that uses snippet variables in its twitter card title and description, see them work.

## Quality assurance


Fixes #10824
